lib/src/nt4_client.dart

```diff
@@ -388,8 +419,20 @@ class NT4Client {
       return;
     }
 
+    _mainWebsocket =
+        WebSocketChannel.connect(Uri.parse(serverAddr), protocols: [
+      'networktables.first.wpi.edu',
+      'v4.1.networktables.first.wpi.edu',
+    ]);
+
+    // Prevents connecting to the wrong address when changing IP address
+    if (!serverAddr.contains(serverBaseAddress)) {
+      return;
+    }
+
```

Comment:
Also, it looks like you have some duplicate code in here from the last PR. These lines for example. There's a few others throughout the changes as well

Reply:
I did the merging stuff wrong (gonna have to reset part of the branch) so all of the commits appear again